app/questionnaire/rules.py

```diff
@@ -36,6 +38,53 @@ def evaluate_rule(when, answer_value):
     return result
 
 
+def evaluate_date_rule(when, answer_store, group_instance, metadata, answer_value):
```

Comment:
Consider refactoring the logic to work out `match_value` into a separate function.

Reply:
Agree with this, then only have evaluation in this method